server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -141,10 +160,23 @@ class ArrivalsGraphStage(initialBaseArrivals: Set[Arrival],
       } else log.info(s"outMerged not available to push")
     }
 
-    def mergeArrivals(base: Set[Arrival], live: Set[Arrival]): Map[Int, Arrival] = {
+    def mergeArrivals(base: Set[Arrival], forecast: Set[Arrival], live: Set[Arrival]): Map[Int, Arrival] = {
       val baseById: Map[Int, Arrival] = base.map(a => (a.uniqueId, a)).toMap
 
-      live.foldLeft(baseById) {
+      val withForecast = forecast.foldLeft(baseById) {
+        case (mergedSoFar, forecastArrival) =>
+          baseById.get(forecastArrival.uniqueId) match {
+            case None =>
+              log.info(s"Forecast arrival ${forecastArrival.IATA} on ${SDate(forecastArrival.Scheduled).toLocalDateTimeString()} not found in base arrivals so ignoring")
+              mergedSoFar
+            case Some(baseArrival) =>
+              val actPax = if (forecastArrival.ActPax > 0) forecastArrival.ActPax else baseArrival.ActPax
```

Comment:
do we want transfer from forecast as well? Does forecast have transfer?

Reply:
Excellent catch! LHR does indeed have transfer pax in the forecast feed